nyaa/api_torznab.py

```diff
@@ -101,71 +112,113 @@ def torznab_caps():
     return render_torznab_caps()
 
 
-def torznab_search():
-    page_size = app.config['RESULTS_PER_PAGE']
+def get_categories_filter(cat):
+    categories = []
+    if cat:
+        for item in cat.split(','):
+            # Parse 1_1 format for compatibility
```

Comment:
I don't understand, yes it skips the loop, but doesn't skip the second if. The `if categories:` call inside the `if cat:` is functionally different if you omit the `if cat:`.
In the original code, if cat = '', then it will return []. If cat isn't empty, but results in an empty list (due to unknown category, for example), then it raises EmptySearchError.
In your suggestion, if cat = '', then it would raise EmptySearchError because that doesn't get short circuited.

Alternative: add as first (two) line(s) `if not cat: return []`, reduces the nesting but preserves the intended behavior, is that more acceptable?